sql: TestMemoryMonitorErrorsDuringBackfillAreRetried failed

sql.TestMemoryMonitorErrorsDuringBackfillAreRetried [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15599265?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15599265?buildTab=artifacts#/) on master @ [77978b485385203b01aec17d51b741874f9c5814](https://github.com/cockroachdb/cockroach/commits/77978b485385203b01aec17d51b741874f9c5814):


```
=== RUN   TestMemoryMonitorErrorsDuringBackfillAreRetried
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/babcb8b872d7785e6f86f1301bfa8e65/logTestMemoryMonitorErrorsDuringBackfillAreRetried11095401
    test_log_scope.go:81: use -show-logs to present logs inline
    schema_changer_test.go:7818: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/babcb8b872d7785e6f86f1301bfa8e65/logTestMemoryMonitorErrorsDuringBackfillAreRetried11095401
--- FAIL: TestMemoryMonitorErrorsDuringBackfillAreRetried (4.91s)
=== RUN   TestMemoryMonitorErrorsDuringBackfillAreRetried/local=true
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    schema_changer_test.go:7816: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql_test/pkg/sql/schema_changer_test.go:7816
        	            				github.com/cockroachdb/cockroach/pkg/testutils/subtest.go:28
        	Error:      	Not equal: 
        	            	expected: 3
        	            	actual  : 2
        	Test:       	TestMemoryMonitorErrorsDuringBackfillAreRetried/local=true
        	Messages:   	not all failure conditions were hit 3
    --- FAIL: TestMemoryMonitorErrorsDuringBackfillAreRetried/local=true (2.00s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestMemoryMonitorErrorsDuringBackfillAreRetried.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39473

--- a/pkg/sql/schema_changer_test.go
+++ b/pkg/sql/schema_changer_test.go
@@ -7812,7 +7812,7 @@ func TestMemoryMonitorErrorsDuringBackfillAreRetried(t *testing.T) {
 		tdb.Exec(t, `ALTER TABLE foo EXPERIMENTAL_RELOCATE SELECT ARRAY[$1], 1`,
 			tc.Server(dataNode).GetFirstStoreID())
 		tdb.Exec(t, `ALTER TABLE foo ADD COLUMN j INT NOT NULL DEFAULT 42`)
-		require.Equalf(t, shouldFail.Load(), int64(2), "not all failure conditions were hit %d", shouldFail.Load())
+		require.GreaterOrEqualf(t, shouldFail.Load(), int64(2), "not all failure conditions were hit %d", shouldFail.Load())
 	})
 }
 

